extensions/binding/org.eclipse.smarthome.binding.bluetooth.bluez/src/main/java/org/eclipse/smarthome/binding/bluetooth/bluez/BlueZBluetoothDevice.java

```diff
@@ -136,6 +138,18 @@ private void enableNotifications() {
             notification.setRssi(n);
             notifyListeners(BluetoothEventType.SCAN_RECORD, notification);
         });
+        device.enableManufacturerDataNotifications(n -> {
+            for (Map.Entry<Short, byte[]> entry : n.entrySet()) {
+                BluetoothScanNotification notification = new BluetoothScanNotification();
+                byte[] data = new byte[entry.getValue().length + 2];
+                data[0] = (byte) (entry.getKey() & 0xFF);
+                data[1] = (byte) (entry.getKey() >> 8);
```

Comment:
don't you need to erase the unwanted bits first by either `& 0xFF00`-ing it before or `& 0xFF`-ing it after the shift?

Reply:
No, because they are dropped by shifting.

Reply:
No, technically they are not all dropped - see https://docs.oracle.com/javase/tutorial/java/nutsandbolts/op3.html. The `>>` operator replicates the highest order bit when shifting. If you want to fill the higher order bits with `0`s, then you'd have to use the `>>>` operator.

Reply:
Thanks for the lesson. That used to be different in C when I did that 30 years ago ;-)